### ISSUE TITLE
depth debug problems

### ISSUE THREAD

**Issue by lollita** [z3APA3A/3proxy#295]
I have a pc with two lan and 3proxy work perfectly.
I have other pc with two lan but 3proxy not work.
it's as if the external ip did not connect to the internet.
How can I investigate this problem further?

**Comment by z3APA3A** [comment]
Check the log for request and error code.
https://3proxy.ru/howtoe.asp#ERRORS

**Comment by lollita** [comment]
I cheched error.
I get a "13 - failed to connect()"
dns with same external work.
same error with all two lan.
Win7.
how to solve it?

**Comment by lollita** [comment]
I have setting:

auth iponly
flush
internal 192.168.1.10
external 192.168.2.10
allow * 192.168.1.10
proxy -p8084

that not work.

if I disable 192.168.1.10
and set

internal 192.168.2.10
external 192.168.2.10

it work.

if I disable 192.168.2.10
and set

internal 192.168.1.10
external 192.168.1.10

it not work.

**Comment by lollita** [comment]
auth iponly
flush
internal 192.168.2.10
external 192.168.2.10
allow * 192.168.2.10
proxy -p8084

work if 192.168.1.10 is disabled
if 192.168.1.10 is enabled together with 192.168.2.10 not work

**Comment by lollita** [comment]
I switched to devel version.

"-b" switch is not present in windows telnet

telnet -b 192.168.2.10 destination_ip destination_port
what for destination_ip destination_port?

>you should either configure source routing for 192.168.2.10 to valid interface
How to ?

or to use -De option (available in devel) to bind to valid interface
Do windows support SO_BINDTODEVICE ?

**Comment by z3APA3A** [comment]
In windows binding to IP always binds to interface

try 
ping -S 192.168.2.10 targethost

**Comment by lollita** [comment]
ping -S 192.168.2.10 www.google.com
work
ping -S 192.168.1.10 www.google.com
request expired

192.168.2.10 have DNS == Gateway
192.168.1.10 have DNS != Gateway

How to solve?

**Comment by z3APA3A** [comment]
This issue is not DNS related and hs no relation to proxy either, this is something with routing. Check you have default gateways for both interfaces.

**Comment by lollita** [comment]
> Check you have default gateways for both interfaces.
how to do?
In setting i have Gateway and outside of 3proxy they work individually both

**Comment by lollita** [comment]
I'm able to work my configuration if i set the lan 192.168.2.10 as default Internet connection.
But why?
If i set
external 192.168.2.10
3proxy should not be that network with its gateway 192.168.2.1 to go on the internet also if it is not the default Internet connection?
There is no way on Windows to make it work even without changing the setting of the default Internet connection?
-De work only on Linux.

**Comment by z3APA3A** [comment]
Sorry, but I have nothing to add here - you have routing issue, it's not proxy specific and it can not be fixed with proxy, In order to use 2 interfaces you have to configure 2 gateways.

**Comment by lollita** [comment]
I apologize, but I ask only to understand.
In my Win7 OS I have configured two LANs and both have a configured gateway.
Both work and connect to the internet.
I just do not understand why 3proxy does not just configure the internet lan with external but also set it in the os as default Internet connection.
Also because if there is already the external parameter, also -De has been added.
it is possible to add support to -De also for windows with the interface number visible in "route print"?
Thanks for your tiny, modular and effective 3proxy.
